hail/src/main/scala/is/hail/io/fs/FS.scala

```diff
@@ -0,0 +1,123 @@
+package is.hail.io.fs
+
+import java.io._
+import java.util
+
+import is.hail.utils.{TextInputFilterAndReplace, WithContext}
+import net.jpountz.lz4.LZ4Compressor
+import com.esotericsoftware.kryo.io.{Input, Output}
+import org.apache.hadoop.fs.FSDataInputStream
+
+trait FileSystem {
+  def open: FSDataInputStream
+  def open(path: FilePath): FSDataInputStream
+  def open(path: String): FSDataInputStream
+
+  def getPath(path: String): FilePath
+  def makeQualified(path: String): FilePath
+  def makeQualified(path: FilePath): FilePath
+  def deleteOnExit(path: FilePath): Boolean
+  // def getFileBlockLocations
+}
+
+trait FilePath extends Serializable{
+  type Configuration
+
+  def toString: String
+  def getName: String
+  def getFileSystem(conf: Configuration): FileSystem
+}
+
+trait FileStatus extends Serializable {
```

Comment:
Need to double check that these need to be serializable

Reply:
I don't see that Hadoop Path is Serializable, but its properties all are. Not sure whether better to take out serializable trait.